server/middleware/registration/index.js

```diff
@@ -0,0 +1,114 @@
+const bcrypt = require('bcrypt');
+const logger = require('simple-node-logger').createSimpleLogger();
+const {
+    HttpStatusCodes: { StatusCodes },
+    setLogDetails,
+    DataException
+} = require('../../utils/constants');
+const mongoUtil = require('../../utils/mongoUtil');
+const { HttpStatusCodes } = require('../../utils/constants');
+const JWT = require('jsonwebtoken');
+const { uuid } = require('uuidv4');
+
+const registrationService = async (req, res) => {
+    let saltRounds = 10;
+    let data, statusCode, collectionObj, userPayload, userId;
+
+    let {
+        firstName,
+        lastName,
+        prefLoc,
+        title,
+        email,
+        userName,
+        password,
+    } = req.body;
+    userId = email;
+
+    password = bcrypt.hashSync(password, saltRounds);
+
+    userPayload = {
+        'email': email,
+        'userId': email,
+        'userName': userName,
+        'password': password,
+        'title': title,
+        'prefLoc': prefLoc,
+        'firstName': firstName,
+        'lastName': lastName,
+    }
+
+    try {
+        collectionObj = req.body.collectionObj ? req.body.collectionObj : await mongoUtil.fetchCollection
+            (process.env.USERS_COLLECTION);
```

Comment:
not really sure why we would have collectionObj in req.body, this is reserved for objects we received in http request (for example, post, put, patch... could also be in get and delete, but rarely). Rewrite to:

`const collectionObj = await mongoUtil.fetchCollection(process.env.USERS_COLLECTION);`

Reply:
resolved in commit # https://github.com/cfernando415/femmecubator-api-repo/pull/36/commits/9829f4621df493aa4c36047ffdb6bbeeeeb87451